Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -37,6 +37,34 @@ TimeSplitter::TimeSplitter(const DateAndTime &start, const DateAndTime &stop) {
   clearAndReplace(start, stop, DEFAULT_VALUE);
 }
 
+/**
+ * Note: The amount of X values in input MatrixWorkspace must be 1 larger than the amount of Y values.
+ * There are NO undefined split regions here.
+ **/
+TimeSplitter::TimeSplitter(const Mantid::API::MatrixWorkspace_sptr &ws, const DateAndTime &offset) {
+  if (ws->getNumberHistograms() != 1) {
+    throw std::runtime_error("MatrixWorkspace can only have 1 histogram when constructing TimeSplitter.");
+  }
+
+  const auto X = ws->binEdges(0);
+  const auto &Y = ws->y(0);
+  if (std::any_of(X.begin(), X.end(), [](double i) { return static_cast<int>(i) < 0; })) {
+    throw std::runtime_error("All X values in MatrixWorkspace must be >= 0 to construct TimeSplitter.");
+  }
+  if (X.size() != Y.size() + 1) {
+    throw std::runtime_error(
+        "Size of x values must be one more than size of y values to construct TimeSplitter from MatrixWorkspace.");
+  }
+
+  int64_t offset_ns{offset.totalNanoseconds()};
+  for (size_t i = 1; i < X.size(); i++) {
+    auto timeStart = Types::Core::DateAndTime((X[i - 1]), 0.0) += offset_ns;
```

Comment:
I think @gecage952 meant `+` instead of `++`;
I have run a simple test:

int a = 5;
int b = 6;
int c = a += b;
int d = a+b;
std::cout << a << "," << b << "," << c << "," << d << std::endl;

The result is:

11,6,11,17

Reply:
I was going for consistency with how Robert was doing it, but now that you mention it, I can just add the offset to the Date and time Constructor directly.

Reply:
I am actually using `+=` on a separate command.  

At first I thought an expression like `c = a += b` wouldn't give the right result in `c`, but it actually does.
So it just depends whether you want to modify `a`, while getting `c`.  In this case, I guess, you don't care, because `a` is a temporary.

Reply:
Gotcha. :sweat_smile: . I just changed it to be clearer as to what's going on.

Reply:
@gecage952 , In the `TableWorkspace` case, I had to be very careful with the `DateAndTime` constructor, because the table holds the times in seconds.  Yes, there is a `DateAndTime` constructor `DateAndTime(const double seconds, const double nanoseconds)`, but it does not work as one might expect.  So in my case I had to create `DateAndTime` from the seconds first and then add the nanosecond offset separately.